backend/views/core/settings/view.py

```diff
@@ -27,8 +29,29 @@ def settings_page(request: HttpRequest):
                 messages.error(request, "Invalid currency")
 
         if profile_picture:
-            usersettings.profile_picture = profile_picture
-            usersettings.save()
+            try:
+                # Max file size is 10MB (Change the first number to determine the size in MB)
+                max_file_size = 10 * 1024 * 1024
+
+                if profile_picture.size <= max_file_size:
+                    img = Image.open(profile_picture)
+                    img.verify()
+
+                    if img.format.lower() in ["jpeg", "png", "jpg"]:
+                        usersettings.profile_picture = profile_picture
+                        usersettings.save()
+                    else:
+                        messages.error(
+                            request,
+                            "Unsupported image format. We support only JPEG, JPG, PNG.",
+                        )
+                else:
+                    messages.error(request, "File size should be up to 10MB.")
+
+            except (FileNotFoundError, Image.UnidentifiedImageError):
+                messages.error(request, "Invalid or unsupported image file")
+        else:
```

Comment:
Lines 53 and 54 give an error while trying to run (python manage.py test backend).
If we delete them the error goes away.